### DISCUSSION TITLE
Can you make kitty send mouse left button events to the program running on it?

### DISCUSSION THREAD

**Discussion by wind0204** [kovidgoyal/kitty#3564] [Q&A] [answered]
Can you make kitty send mouse click/drag events to the program running on it when the part clicked on doesn't have any URL or hyperlink when user clicked/dragged? kitty didn't share the events of mouse left button clicks and drags but only of mouse right button events with vim.

I've recently started using kitty and I love many of the powerful features in it and I think there must be an easy way to prevent kitty from stealing the mouse events.

I'm currently using this version: ae43b1565e6a678404c910b1e00d29141d6c4de6 (a grand-grandchild of v0.20.2)

**Comment by kovidgoyal** [comment]
It does already if the program running in the terminal requests it too

**Reply by wind0204** [reply]
It seems to me though vim does ask for the events including the left button events since it works fine on xfce4-terminal.

**Comment by kovidgoyal** [comment]
And it works fine in kitty too. 

set mouse=a

in your vimrc

On Sun, May 02, 2021 at 03:12:07AM -0700, Nicholas Gim wrote:
> It seems to me though vim does ask for the events including the left button events since it works fine in xfce4-terminal.
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/3564#discussioncomment-684612

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Reply by wind0204** [reply]
I wonder why it doesn't work in my kitty, vim writes `a` when I type `:echo &mouse`

**Comment by wind0204** [comment, marked as answer]
It's been a kind of busy week for me, I just checked the git log now and saw a bunch of commits about mouse events, rebuilt kitty and.. now it does work! perfect!

I'm gonna check out some of the new features added in. 😊
